runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CoderTranslationTest.java

```diff
@@ -152,14 +150,12 @@ public static Iterable<Coder<?>> data() {
               StringUtf8Coder.of(),
               SerializableCoder.of(Record.class),
               new RecordCoder(),
-              KvCoder.of(
```

Comment:
Can we preserve the test with AvroCoder here? It sounds fine to me that runners-core-construction depend on the avro module (even though main jar doesn't).

EDIT: I see it may introduce circular dependency. How about add a test like originally this one does in the extensions/avro module? like sdks/java/extensions/avro/src/test/java/org/apache/beam/sdk/extensions/avro/AvroCoderTranslationTest.java

Reply:
Yes it introduces a circular dependency. I split the test case into `AvroCoderTranslationTest` already, no? I did not keep the KV since I don't think we need to test that we can correctly serialize nested coders - that is tested in lots of places and does not depend on what the nested coder is.

Is there a specific test case or cases you want me to add to `AvroCoderTranslationTest`? I am happy to do it but I'm not sure the most important scope (right now it just tests the coder from line 157).

Reply:
> I split the test case into AvroCoderTranslationTest already

Got it. AvroCoderTranslationTest specifically test the line "AvroCoder.of(..." and below